src/v/cluster/cloud_metadata/uploader.cc

```diff
@@ -246,6 +250,12 @@ ss::future<error_outcome> uploader::maybe_upload_controller_snapshot(
 }
 
 ss::future<> uploader::upload_until_abort() {
+    // It's possible the cluster UUID wasn't available at construction time
+    // (e.g. this node was in the process of joining the cluster).
+    if (!co_await _storage.wait_for_cluster_uuid()) {
+        co_return;
+    }
```

Comment:
nit: could check that `cluster_uuid == model::cluster_uuid{}` first.

Reply:
Done